src/main/java/com/snowplowanalytics/snowplow/tracker/payload/TrackerPayload.java

```diff
@@ -40,11 +40,12 @@
     @Override
     public void add(final String key, final String value) {
         if (key == null || key.isEmpty()) {
-            LOGGER.error("Invalid key detected: {}", key);
+            LOGGER.error("Null or empty key detected");
             return;
         }
         if (value == null || value.isEmpty()) {
-            LOGGER.info("null or empty value detected: {}->{}", key, value);
+            LOGGER.info("Null or empty value detected");
+            LOGGER.debug("Null or empty value detected: {}->{}", key, value);
```

Comment:
I'm going to be a pain and after seeing this, suggest we just keep the debug log. In reality, it's ok that we strip these null or empty values out of the payload, and the info log feels like it might be a bit noisy if someone is trying to send nulls or empty strings in a legit use case (maybe because it's just easier to code it that way in their application). So lets just keep the debug log and remove the info line.

Reply:
👍